### ISSUE TITLE
`dart test` doesn't recognize the `--define` or '-D' flag

### ISSUE THREAD

**Issue by rxlabz** [dart-lang/sdk#45827]
even if #44562 is closed, it seems actually not possible to define environment variables fort the `dart test` command ?

**Comment by lrhn** [comment]
The flag should be named `--define` or `-D`, not `--dart-define` (that's Flutter's name, Dart itself doesn't need to say that).

Have you tried `dart -Dfoo=bar test`?

**Comment by rxlabz** [comment]
yes also tried with `-D` but the result is `Could not find an option with short name "-D".`

**Comment by rxlabz** [comment]
the workaround seems to be : 

```bash
export DART_VM_OPTIONS='-Dfoo=bar' && \                                                              1 ↵ ✹ ✚ ✭
          dart test
```

**Comment by bkonyi** [comment]
cc @natebosch @grouma

**Comment by natebosch** [comment]
`dart -Dfoo=true test` will do what you are hoping.

Importantly pass the `-D` after `dart`, not after `test` in the command.

**Comment by bkonyi** [comment]
`dart test -Dfoo=true` should also be valid, as that support has been added for other commands (see [dart run](https://github.com/dart-lang/sdk/blob/master/pkg/dartdev/lib/src/commands/run.dart#L112) and the various [dart compile commands](https://github.com/dart-lang/sdk/blob/master/pkg/dartdev/lib/src/commands/compile.dart#L92)).
